library/src/CMakeLists.txt

```diff
@@ -205,11 +205,16 @@ rocm_export_targets(
 
 rocm_install_symlink_subdir( rocblas )
 
+#Control for enabling or disabling header testing
+option (RUN_HEADER_TESTING ON)
+
+if(BUILD_TESTING AND RUN_HEADER_TESTING)
```

Comment:
Why not just set `RUN_HEADER_TESTING` default based on `BUILD_TESTING`: `option (RUN_HEADER_TESTING "" ${BUILD_TESTING})`? Then you can just check `RUN_HEADER_TESTING` here instead of both.